modules/deploy/main.tf

```diff
@@ -106,7 +109,7 @@ resource "local_file" "deploy_script" {
 }
 
 resource "null_resource" "deploy" {
-  count = var.create && var.create_deployment ? 1 : 0
+  count = var.create && var.create_deployment && var.run_deployment ? 1 : 0
```

Comment:
Looks good. Could you also update related examples in the `examples/deploy` directory and verify them?